tools/protodoc/protodoc.py

```diff
@@ -240,10 +240,14 @@ def FormatMessageAsJson(type_context, msg):
   Return:
     RST formatted pseudo-JSON string representation of message definition.
   """
-  lines = [
-      '"%s": %s' % (f.name, FormatFieldTypeAsJson(type_context, f))
-      for f in msg.field
-  ]
+  lines = []
+  for index, field in enumerate(msg.field):
+    field_type_context = type_context.Extend([2, index], field.name)
```

Comment:
Not specific to this PR, but I'm getting a bit icked out by my use of path magic numbers in protodoc.py, I should clean this up at some point.